os/macosx/notarize.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -e
+
+# This script attempts to notarize the OpenTTD DMG generated by cpack.
+# If you are building an unofficial branch of OpenTTD, please change the bundle
+# ID in Info.plist and below.
+#
+# This uses `gon' to perform notarization:
+#
+#   https://github.com/mitchellh/gon
+#
+# Follow the setup instructions on the gon site to install.
+#
+# Before executing this script, you must first configure CMake with at least the following
+# parameters:
+#
+# -DCPACK_BUNDLE_APPLE_CERT_APP={certificate ID}
+# "-DCPACK_BUNDLE_APPLE_CODESIGN_PARAMETER=--deep -f --options runtime"
+#
+# then run "make package" or "cpack".
+#
+# This will sign the application with your signing certificate, and will enable
+# the hardened runtime.
+#
+# You also need to set your Apple Developer username and password (app-specific password
+# is recommended) in the AC_USERNAME and AC_PASSWORD environment variables.
+#
+# Then, ensuring you're in your build directory and that the "bundles" directory
+# exists with a .dmg in it (clear out any old DMGs first), run:
+#
+# ../os/macosx/notarize.sh
+
+if [ -z "${AC_USERNAME}" ]; then
+    echo AC_USERNAME not set, skipping notarization.
+    exit 0
+fi;
+
+dmg_filename=(bundles/*.dmg)
+
+if [ "${dmg_filename}" = "bundles/*.dmg" ]; then
```

Comment:
```
dmg_filename=$(ls bundles/*.dmg)

if [ -z "${dmg_filename}" ]; then ...
```

Might look a bit better.

Reply:
That results in:

`ls: bundles/*.dmg: No such file or directory`

Taking out the `set -e` gives the same error followed by our custom error. Although it's not the most elegant, the code that's there seems to work without too much fuss.

Reply:
Oops:

`$(ls bundles/*.dmg 2>/dev/null || true)`

Reply:
Nope, that doesn't then split the string up into an array if multiple DMGs happen to be in the folder. I'm sure there will be some further way around that, but it feels it's getting a bit more convoluted again. :)